floris/simulation/base.py

```diff
@@ -37,13 +43,16 @@ class State(Enum):
     USED = 2
 
 
-class BaseClass(LoggerBase, FromDictMixin):
+@define
+class BaseClass(FromDictMixin):
     """
     BaseClass object class. This class does the logging and MixIn class inheritance.
     """
 
-    state = State.UNINITIALIZED
-
+    # Initialize `state` and ensure it is treated as an attribute rather than a constant parameter.
+    # See https://www.attrs.org/en/stable/api-attr.html#attr.ib
+    state = field(init=False, default=State.UNINITIALIZED)
+    _logging_manager: LoggingManager = field(init=False, default=LoggingManager())
 
     @classmethod
     def get_model_defaults(cls) -> Dict[str, Any]:
```

Comment:
In `get_model_defaults` or somewhere else? Sorry, I don't follow, can you give an example?

Reply:
Essentially what will happen in `get_model_defaults` is that it will also provide the "state" and "_logging_manager" values (toy example below).

![image](https://github.com/NREL/floris/assets/13874373/af84248b-7428-4ece-a790-698fba4c28da)

In the below toy example, we can see the `FromDictMixin.as_dict()` method doesn't provide an internally initialized value that can be used to create a new model.

![image](https://github.com/NREL/floris/assets/13874373/2827bda8-a7ff-4443-aa58-d5ecbd03c4ad)

However, the use case is different, so `get_model_defaults` could just have a filter for `if el.init` to produce working inputs (example below).

![image](https://github.com/NREL/floris/assets/13874373/bebadc1e-8548-4214-8c71-debc3d854ba9)

Reply:
As a quick note/clarification to the second example, a parameter defined without a default will cause that method to fail if it hasn't yet been initialized. That's why I added a default in that example, and not in the others, but forgot to make that explicit.

Reply:
Well, `get_model_defaults` is not used in FLORIS, so I propose to remove this in favor of using the `Model.as_dict()` function. As I understand, that is part of what you're pointing out, anyway. What do you think @RHammond2?

Reply:
(I made this change and appended it to the other commit that removed `_get_model_dict` since this is all related to using `as_dict` instead of a redundant method)